app/controllers/ProvideConfidentialInformationController.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import connectors.DataCacheConnector
+import controllers.actions._
+import config.FrontendAppConfig
+import forms.ProvideConfidentialInformationFormProvider
+import models.Mode
+import pages.{ProvideConfidentialInformationPage, ProvideGoodsNamePage}
+import navigation.Navigator
+import play.api.mvc.MessagesControllerComponents
+import views.html.provideConfidentialInformation
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ProvideConfidentialInformationController @Inject()(
+                                        appConfig: FrontendAppConfig,
+                                        dataCacheConnector: DataCacheConnector,
+                                        navigator: Navigator,
+                                        identify: IdentifierAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        formProvider: ProvideConfidentialInformationFormProvider,
+                                        val provide_confidential_information: views.html.provideConfidentialInformation,
+                                        cc: MessagesControllerComponents
+                                     )(implicit ec:ExecutionContext) extends FrontendController(cc) with I18nSupport {
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode) = (identify andThen getData andThen requireData) {
+    implicit request =>
+      val goodsName = request.userAnswers.get(ProvideGoodsNamePage).getOrElse("")
```

Comment:
I wonder what is the best default here - should we default it to `.getOrElse("goods")` as per the AddConfidentialInformationController?

In ProvideGoodsDescriptionController there is no default and it would throw an exception. It might be worth checking all of these places and using the same default, maybe we could even move this logic into a Helper class or something?

Reply:
I think the `getOrElse("goods")` is probably the best approach as the page text still makes sense that way 👍

Reply:
this has now been addressed